kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/EnvelopeEncryptionST.java

```diff
@@ -54,6 +55,10 @@ void setUp() {
         else {
             NamespaceUtils.createNamespaceWithWait(Vault.VAULT_DEFAULT_NAMESPACE);
             kubeVaultTestKmsFacade.start();
+            assertThat(kubeVaultTestKmsFacade.isCorrectVersionInstalled())
```

Comment:
I'd suggest giving KubeVaultTestKmsFacade#startVault() the responsibility to check whether the expected version is install, if no, throw an exception.  It shouldn't really be a concern of the tests.  isCorrectVersionInstalled is then private.